saas/managers/metrics.py

```diff
@@ -40,10 +40,11 @@ def month_periods(nb_months=12, from_date=None):
     if not from_date:
         # By default, we pick tomorrow so that income from Today shows up.
         from_date = datetime.utcnow().replace(tzinfo=utc) + timedelta(days=1)
-    if isinstance(from_date, basestring):
-        from_date = datetime.strptime(from_date, '%Y-%m')
-    from_date = datetime(day=from_date.day, month=from_date.month,
-        year=from_date.year, tzinfo=utc)
+    else:
```

Comment:
Incorrect (and unnecessary) change. It introduce two subtle errors:
1. If from_date exists but is not a basestring, it might lead to a 500 error farther down the line that it used to.
2. from_date is no more normalized to midnight utc in every single case.